File: 3391_design_a_3d_binary_matrix_with_efficient_layer_tracking/Solution.cs
// LeetCode 3391 - Design a 3D Binary Matrix with Efficient Layer Tracking
// https://leetcode.com/problems/design-a-3d-binary-matrix-with-efficient-layer-tracking/

public class Matrix3D {
    int[][][] m;
    int[] ones;
    int n;

    public Matrix3D(int n) {
        this.n = n;
        m = new int[n][][];
        ones = new int[n];
        for (int i = 0; i < n; i++) {
            m[i] = new int[n][];
            for (int j = 0; j < n; j++) m[i][j] = new int[n];
        }
    }

    public void SetCell(int x, int y, int z) {
        if (m[x][y][z] == 0) {
            m[x][y][z] = 1;
            ones[x]++;
        }
    }

    public void UnsetCell(int x, int y, int z) {
        if (m[x][y][z] == 1) {
            m[x][y][z] = 0;
            ones[x]--;
        }
    }

    public int LargestMatrix() {
        int best = -1, idx = 0;
        for (int i = 0; i < n; i++) {
            if (ones[i] >= best) {
                best = ones[i];
                idx = i;
            }
        }
        return idx;
    }
}

File: 3391_design_a_3d_binary_matrix_with_efficient_layer_tracking/solution.php
<?php
// LeetCode 3391 - Design a 3D Binary Matrix with Efficient Layer Tracking
// https://leetcode.com/problems/design-a-3d-binary-matrix-with-efficient-layer-tracking/

class Matrix3D {
    public $n;
    public $m;
    public $ones;

    function __construct($n) {
        $this->n = $n;
        $this->m = [];
        for ($x = 0; $x < $n; $x++) {
            $this->m[$x] = [];
            for ($y = 0; $y < $n; $y++) $this->m[$x][$y] = array_fill(0, $n, 0);
        }
        $this->ones = array_fill(0, $n, 0);
    }

    function setCell($x, $y, $z) {
        if ($this->m[$x][$y][$z] === 0) {
            $this->m[$x][$y][$z] = 1;
            $this->ones[$x]++;
        }
    }

    function unsetCell($x, $y, $z) {
        if ($this->m[$x][$y][$z] === 1) {
            $this->m[$x][$y][$z] = 0;
            $this->ones[$x]--;
        }
    }

    function largestMatrix() {
        $best = -1;
        $idx = 0;
        for ($i = 0; $i < $this->n; $i++) {
            if ($this->ones[$i] >= $best) {
                $best = $this->ones[$i];
                $idx = $i;
            }
        }
        return $idx;
    }
}
